internal/utils/utils.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"sort"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type setFn = func(d *schema.ResourceData, key string) error
+
+// setMap sets the map of values to ResourceData, checking and returning the
+// errors. Typically d.Set is not error checked. This helper makes checking
+// those errors less tedious. Because this works with a map, the order of the
+// errors would not be predictable, to avoid this the errors will be sorted.
+func SetMap(d *schema.ResourceData, m map[string]interface{}) error {
```

Comment:
Not in scope for this PR, but looking at how and where this helper is used, I'm not convinced it's really necessary.

Reply:
The pattern SetMap was made to simplify was 

```
d.Set("foo", foo)
d.Set("bar", bar)
```

Which is repetitive and common, but is also bad practice.

`d.Set` can error out and those errors were seldom checked. When they are checked, you end up with resources with 10-30 lines per attributes and an extra 20-60 lines of `if {`, `}`.